src/xrc/src/lib.rs

```diff
@@ -1505,4 +1505,48 @@ mod test {
         assert!(matches!(
             btt_btc_exchange_rate.validate(), Ok(rate) if rate.decimals == Some(11)));
     }
+
+    #[test]
+    /// The function verifies that the [QueriedExchangeRate] struct can be inverted correctly despite
+    /// having been provided an empty rates array.
+    fn inverting_queried_exchange_rate_with_no_rates() {
+        let rate = QueriedExchangeRate::new(btc_asset(), usd_asset(), 0, &[], 0, 0, None);
+
+        assert!(matches!(
+            rate.inverted().validate(),
+            Err(ExchangeRateError::Other(_))
```

Comment:
What is the actual error?

Reply:
Updated the test to check code and message.